demo/src/features/debug3d/mod.rs

```diff
@@ -1,53 +1,71 @@
-use crate::features::debug3d::extract::Debug3dExtractJob;
-use rafx::framework::{VertexDataLayout, VertexDataSetLayout};
-use rafx::nodes::ExtractJob;
-use rafx::nodes::RenderFeature;
-use rafx::nodes::RenderFeatureIndex;
-use std::convert::TryInto;
-
-mod debug3d_resource;
+rafx::declare_render_feature_mod!();
+rafx::declare_render_feature_renderer_plugin!();
+
+rafx::declare_render_feature!(Debug3DRenderFeature, DEBUG_3D_FEATURE_INDEX);
+
 mod extract;
-mod plugin;
 mod prepare;
 mod write;
-pub use plugin::Debug3DRendererPlugin;
 
-pub use debug3d_resource::*;
-use rafx::api::RafxPrimitiveTopology;
+mod public;
```

Comment:
if something is public I would just leave it in the root of the feature's module.

Reply:
I think the template is easier to understand with the `public` folder because you can re-export that entire directory inside of `mod.rs` with just `pub use public::*`, but you can't do that inside of `mod.rs` without it.

Reply:
Looking at the features we have now, only 1/6 of them have more than one module that we would re-export publicly. So I think we will end up with a lot of public folders containing an almost empty mod.rs and a single .rs file with the public-facing type.

If anything, I would suggest have a jobs folder and moving the extract/prepare/write into that and leaving the other public modules in the root. I don't mind an "internal" module/folder as much but I think it too will rarely contain more than 1 file. In general, I prefer a flat structure unless there are >3 files that can be grouped into a directory.

Would having a jobs directory and leaving the public modules in the root feel more "readable" to you? Then you could do pub use crate::feature_name::* and get all the public symbols.

Reply:
So the proposed directory would be something like:

```
.\feature_name\
    .\internal\
        .\<optional helper code>.rs
    .\jobs\
        extract.rs
        prepare.rs
        write.rs
    .\mod.rs
    .\plugin.rs
    .\<optional top level interfaces>.rs
```

`internal` wouldn't be present unless it was actually needed. `jobs` would always exist. There would be no more `public` directory, because anything at the top-level _is_ public.

I can be fine with that.

Reply:
yeah, I'm find with this too!